### ISSUE TITLE
Role='application' around ChoiceGroup; bug or feature?

### ISSUE THREAD

**Issue by eivindland** [OfficeDev/office-ui-fabric-react#6479]
Around ChoiceGroup there is a wrapper with role='application'. When testing for accessibility-issues my screen reader software reads this as _"Your are in an application"_. 

The radio buttons are then read correctly, but I wonder if the role really is necessary? Every document I've found online says something like: _"Use role='application' with care"._

I cannot find a way to remove or disable the role='application' from the ChoiceGroup component.

**Comment by KevinTCoughlin** [comment]
👋 @eivindland 

You're correct that `ChoiceGroup` does not currently allow its `role="application"` to be overridden. We can easily make that a prop cc: @micahgodbolt. Here's the LoC:

https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.base.tsx#L79

I tried to track down the history of that attribute, without luck. Reading the ARIA spec on `aria-role="application"` https://www.w3.org/TR/wai-aria-1.1/#application:

>A structure containing one or more focusable elements requiring user input, such as keyboard or gesture events, that do not follow a standard interaction pattern supported by a widget role.

I'm not sure any of the other `widget` roles satisfy the interaction model with `ChoiceGroup`. I think our reliance on it is mostly due to its makeup of `<div>` rather than a radio group (for example). But I'm not sure its usage is incorrect here.

Just curious, what would you override it to?

**Comment by eivindland** [comment]
Thank your for replying so quickly! I'm also not entirely sure that the usage of role=application is incorrect here. But to me it seems that the intention of this role is to support custom elements, like widgets, and to tell the screen reader to change the mode on how the keyboard commands are interpreted. 

[http://accessibleculture.org/articles/2011/02/not-all-aria-widgets-deserve-role-application/](http://accessibleculture.org/articles/2011/02/not-all-aria-widgets-deserve-role-application/)
> Using role="application" tells supporting assistive technologies to change their mode of behaviour. Instead of intercepting keyboard commands to allow interaction with the virtual buffer, they pass all keyboard input through to the browser as if dealing with an application. This goes for all of the content wrapped in a container with this role, whether it's made of up explicit form controls or just simple text.

Some users report that they do not like when the screen reader is forced to act in a certain way, and they cannot use their standard keyboard shortcuts.
So I think it is a great idea to make it a prop. I would like to be able to drop the role=application completely, and just rely on role=radiogroup in the wrapping div. (I believe this exists already)

**Comment by KevinTCoughlin** [comment]
Raised a PR since the change was very easy, but left my comment re: my concern in the description. https://github.com/OfficeDev/office-ui-fabric-react/pull/6495

**Comment by jamesedjac** [comment]
In this case I would actually advise that role="application" be removed entirely as this use does not conform to the [ARIA 1.1 standard](https://www.w3.org/TR/wai-aria-1.1/#application), which states

> A structure containing one or more focusable elements requiring user input, such as keyboard or gesture events, that _do not follow a standard interaction pattern supported by a widget role_.

In this case, choice group follows the [Radio Group](https://www.w3.org/TR/wai-aria-practices-1.1/#radiobutton) widget role.

Also in general, role="application" should not be considered "stable" as it causes problems with many different Screen reader and browser combinations. In addition, it is not necessary because most screen readers already honor developer defined keyboard behavior even without role="application" declared

**Comment by KevinTCoughlin** [comment]
> 
> 
> In this case I would actually advise that role="application" be removed entirely as this use does not conform to the [ARIA 1.1 standard](https://www.w3.org/TR/wai-aria-1.1/#application), which states
> 
> > A structure containing one or more focusable elements requiring user input, such as keyboard or gesture events, that _**do not** follow a standard interaction pattern supported by a widget role_.
> 
> In this case, choice group follows the [Radio Group](https://www.w3.org/TR/wai-aria-practices-1.1/#radiobutton) widget role.
> 
> Also in general, role="application" should not be considered "stable" as it causes problems with many different Screen reader and browser combinations, and is usually not necessary because most screen readers already honor developer defined keyboard behavior even without role="application" declared

Thank you for clarifying the usage here and path forward @jamesedjac!

Would it be acceptable if I merge #6495 that makes this configurable via prop to unblock @eivindland? Then I'll follow-up and see if the JAWS issue is fixed so that we can move towards removing `role="application"` here completely.

**Comment by KevinTCoughlin** [comment]
@jamesedjac next published version of OUIFR will allow you to override the role. I'll work to see if we can remove it completely dependent on the JAWS bug linked in-line. Will keep this open for tracking that.

**Comment by mikewheaton** [comment]
The PR above (#6495) was released in version 6.83.0, so I'm going to go ahead and close this issue.
